src/Compilers/VisualBasic/Portable/Binding/Binder_Invocation.vb

```diff
@@ -3028,87 +3028,23 @@ ProduceBoundNode:
-
-                                            If propertyOrEvent IsNot Nothing Then
-                                                container = propertyOrEvent
-                                            End If
-
-                                            Exit While
-                                        End If
-
-                                    Case Else
-                                        container = container.ContainingSymbol
-                                End Select
-                            End While
 
-                            If container IsNot Nothing AndAlso container.Name IsNot Nothing Then
-                                callerInfoValue = ConstantValue.Create(container.Name)
-                            End If
-                        Else
-                            Debug.Assert(isCallerFilePath)
-                            callerInfoValue = ConstantValue.Create(callerInfoOpt.SyntaxTree.GetDisplayPath(callerInfoOpt.Span, Me.Compilation.Options.SourceReferenceResolver))
-                        End If
-
-                        If callerInfoValue IsNot Nothing Then
-                            ' Use the value only if it will not cause errors.
-                            Dim ignoreDiagnostics = DiagnosticBag.GetInstance()
-                            Dim literal As BoundLiteral
-
-                            If callerInfoValue.Discriminator = ConstantValueTypeDiscriminator.Int32 Then
-                                literal = New BoundLiteral(syntax, callerInfoValue, GetSpecialType(SpecialType.System_Int32, syntax, ignoreDiagnostics))
-                            Else
-                                Debug.Assert(callerInfoValue.Discriminator = ConstantValueTypeDiscriminator.String)
-                                literal = New BoundLiteral(syntax, callerInfoValue, GetSpecialType(SpecialType.System_String, syntax, ignoreDiagnostics))
-                            End If
-
-                            Dim convertedValue As BoundExpression = ApplyImplicitConversion(syntax, param.Type, literal, ignoreDiagnostics)
-
-                            If Not convertedValue.HasErrors AndAlso Not ignoreDiagnostics.HasAnyErrors Then
-                                ' Dev11 #248795: Caller info should be omitted if user defined conversion is involved.
-                                If Not (convertedValue.Kind = BoundKind.Conversion AndAlso (DirectCast(convertedValue, BoundConversion).ConversionKind And ConversionKind.UserDefined) <> 0) Then
-                                    defaultConstantValue = callerInfoValue
-                                End If
-                            End If
+            If defaultConstantValue Is Nothing Then
+                If param.HasExplicitDefaultValue = False Then
+                    If InternalSyntax.Parser.CheckFeatures(InternalSyntax.Feature.ImplicitDefaultValueOnOptionalParameter, _compilation.Options.ParseOptions) Then
```

Comment:
Each SyntaxTree has its own ParseOptions, the check shouldn't be based on them instead.

---
In reply to: [74482094](https://github.com/dotnet/roslyn/pull/12629#discussion_r74482094) [](ancestors = 74482094)